GoogleSignIn/Sources/GIDSignIn.m

```diff
@@ -636,50 +672,42 @@ - (void)addDecodeIdTokenCallback:(GIDAuthFlow *)authFlow {
     }
     OIDIDToken *idToken =
         [[OIDIDToken alloc] initWithIDTokenString:authState.lastTokenResponse.idToken];
+    // If the picture and name fields are present in the ID token, use them, otherwise make
+    // a userinfo request to fetch them.
     if (idToken) {
-      // If the picture and name fields are present in the ID token, use them, otherwise make
-      // a userinfo request to fetch them.
-      if (idToken.claims[kBasicProfilePictureKey] &&
-          idToken.claims[kBasicProfileNameKey] &&
-          idToken.claims[kBasicProfileGivenNameKey] &&
-          idToken.claims[kBasicProfileFamilyNameKey]) {
-        handlerAuthFlow.profileData = [[GIDProfileData alloc]
-            initWithEmail:idToken.claims[kBasicProfileEmailKey]
-                     name:idToken.claims[kBasicProfileNameKey]
-                givenName:idToken.claims[kBasicProfileGivenNameKey]
-               familyName:idToken.claims[kBasicProfileFamilyNameKey]
-                 imageURL:[NSURL URLWithString:idToken.claims[kBasicProfilePictureKey]]];
-      } else {
-        [handlerAuthFlow wait];
-        NSURL *infoURL = [NSURL URLWithString:
-            [NSString stringWithFormat:kUserInfoURLTemplate,
-                [GIDSignInPreferences googleUserInfoServer],
-                authState.lastTokenResponse.accessToken]];
-        [self startFetchURL:infoURL
-                    fromAuthState:authState
-                      withComment:@"GIDSignIn: fetch basic profile info"
-            withCompletionHandler:^(NSData *data, NSError *error) {
-          if (data && !error) {
-            NSError *jsonDeserializationError;
-            NSDictionary<NSString *, NSString *> *profileDict =
-                [NSJSONSerialization JSONObjectWithData:data
-                                                options:NSJSONReadingMutableContainers
-                                                  error:&jsonDeserializationError];
-            if (profileDict) {
-              handlerAuthFlow.profileData = [[GIDProfileData alloc]
-                  initWithEmail:idToken.claims[kBasicProfileEmailKey]
-                           name:profileDict[kBasicProfileNameKey]
-                      givenName:profileDict[kBasicProfileGivenNameKey]
-                     familyName:profileDict[kBasicProfileFamilyNameKey]
-                       imageURL:[NSURL URLWithString:profileDict[kBasicProfilePictureKey]]];
-            }
-          }
-          if (error) {
-            handlerAuthFlow.error = error;
+      handlerAuthFlow.profileData = [self profileDataWithIDToken:idToken];
+    }
+      
+    if (!handlerAuthFlow.profileData) {
```

Comment:
Nice refactoring.  Maybe add a comment here to explain what's happening?

Reply:
Done.